core/domain/collection_services.py

```diff
@@ -574,6 +576,29 @@ def get_collection_snapshots_metadata(collection_id):
         collection_id, version_nums)
 
 
+def _get_contributor_ids_for_collection(col_id):
```

Comment:
I don't think it should be too difficult either :-) I would suggest finishing off the other PR, since you're so close (and adding all the tests, still; this is an important calculation and the more aspects of it you can test, the better). We can then do the update for the collection summary contributors in a separate PR. Does this sound good?